codec-dns/src/main/java/io/netty/handler/codec/dns/AbstractDnsMessage.java

```diff
@@ -379,9 +379,10 @@ public DnsMessage retain(int increment) {
     protected void deallocate() {
         clear();
 
-        final ResourceLeak leak = this.leak;
+        final ResourceLeakTracker<DnsMessage> leak = this.leak;
         if (leak != null) {
-            leak.close();
+            boolean closed = leak.close(this);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "closed". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)